app/controllers/ideas_controller.rb

```diff
@@ -43,7 +43,10 @@ def create
 
   # PATCH/PUT /ideas/1
   def update
-    if @idea.update(idea_params)
+    @idea.review_year = idea_params[:review_year]
+    @idea.review_month = idea_params[:review_month]
+    @idea.review_day = idea_params[:review_day]
+    if @idea.update(idea_params) && @idea.errors[:review_date].none?
```

Comment:
When you call `@idea.update` it will run validations and return `false` if they fail. So the second check shouldn't be needed. :)